src/PostgREST/QueryBuilder.hs

```diff
@@ -322,6 +321,22 @@ requestToQuery schema _ (DbMutate (Delete mainTbl logicForest returnings)) =
 ignoredBody :: SqlFragment
 ignoredBody = "ignored_body AS (SELECT $1::text) "
 
+-- |
+-- These CTEs convert a json object into a json array, this way we can use json_populate_recordset for all json payloads
+-- Otherwise we'd have to use json_populate_record for json objects and json_populate_recordset for json arrays
+-- We do this in SQL to avoid processing the JSON in application code
+normalizedBody :: SqlFragment
+normalizedBody =
+  unwords [
+    "pgrst_payload AS (SELECT $1::json AS json_data),",
+    "pgrst_body AS (",
+      "SELECT json_data AS val FROM pgrst_payload WHERE json_typeof(json_data) = 'array'",
+      "UNION ALL",
```

Comment:
could this be written in form of `CASE`?, it would be easier to understand.

Reply:
another benefit would be that `json_type` would be called only once on this big payload

Reply:
CTE is much simpler now, thanks.